main.js

```diff
@@ -0,0 +1,310 @@
+      locationLongitude = position.coords.longitude
+      getLocationAddress(locationLatitude, locationLongitude)
+    });
+  } else {
+    alert('location services not available')
+  }
+})
+
+//on search get location information and pass it to the weather APIs
+$('.search').on('click', function() {
+  input = $('#search-query').val();
+  //show a spinner while fetching data from the API
+  $('.weekly-forecast').addClass('hide')
+  $('.daily-forecast').addClass('hide')
+  $('.spinner').removeClass('hide')
+  //pass location query and false because we are not defining this as the default location
+  getLocationCoordinates(input, false)
+})
+
+//set a default location and save it in localStorage, by passing this location to getLocationCoordinates with boolean true we set it as the default location
+$('body').on("click", ".default-location", function() {
+  var input = $('#search-query').val();
+  localStorage.setItem('location', JSON.stringify(input))
+  getLocationCoordinates(input, true)
+})
+
+//used in geolocation: gets a location name from the location address data to pass to the getLocationCoordinates function to get weather data
+var parseLocation = function(data) {
+  //iterates through location address data to find the item that gives the locality or sublocality name to pass to the getLocationCoordinates function. If no name, alerts an error and hides the spinner
+  var locationName
+  for (i = 0; i < data.results.length; i++) {
+    for (j = 0; j < data.results[i].types.length; j++) {
+      if (data.results[i].types[j] == "locality" || data.results[i].types[j] == "sublocality") {
+        locationName = data.results[i].formatted_address;
+        break;
+      }
+    }
+  }
+  if (!locationName) {
+    $('.spinner').addClass('hide')
+    alert("Geolocation error, please try again.")
+  }
+  $('#search-query').val(locationName)
+  getLocationCoordinates(locationName, false)
+}
+//pass search query to a Google maps API to get full name, latitude, and longitude then pass latitude and longitude to the openweather API to get weather data
+var getWeather = function(data, isDefault) {
+  $('.weekly-forecast').removeClass('hide')
+  $('.daily-forecast').removeClass('hide')
+  $('.spinner').addClass('hide')
+  //if we're getting weather for the default location, we set the default location name
+  if (isDefault) {
+    defaultLocationName = data.results[0].formatted_address
+  }
+  var locationName = data.results[0].formatted_address
+  //checks whether a "normal search" is for the default location - if so we hide the active default location button and show the disabled default location button
+  if (locationName == defaultLocationName) {
+    $('.not-default').addClass('hide');
+    $('.is-default').removeClass('hide');
+  } else {
+    $('.not-default').removeClass('hide');
+    $('.is-default').addClass('hide');
+  }
+  //remove the ', USA' from cities in the United States to improve display
+  if (locationName.search(/,\sUSA/) != -1) {
```

Comment:
Really, you're just relying on the fact that the location name doesn't end in USA, rather than checking the country value (which would be better).